File: repos_estudiantes_ALVAREZ_tercera_entrega/CHINO_BLANCO_NOEMI/Generacidad/Ejercicio3/Pila.py
class Pila:
    def __init__(self):
        self.elementos = []

    def apilar(self, elemento):
        self.elementos.append(elemento)

    def desapilar(self):
        if self.esta_vacia():
            return None
        return self.elementos.pop()

    def mostrar(self):
        print("Contenido de la pila:")
        for elemento in reversed(self.elementos):
            print(elemento)

    def esta_vacia(self):
        return len(self.elementos) == 0

def pruebas():
    print("Pila de enteros")
    pila_enteros = Pila()
    pila_enteros.apilar(10)
    pila_enteros.apilar(20)
    pila_enteros.apilar(30)
    pila_enteros.mostrar()
    print("Desapilar:", pila_enteros.desapilar())
    pila_enteros.mostrar()

    print("\nPila de cadenas")
    pila_cadenas = Pila()
    pila_cadenas.apilar("Hola")
    pila_cadenas.apilar("Mundo")
    pila_cadenas.mostrar()
    print("Desapilar:", pila_cadenas.desapilar())
    pila_cadenas.mostrar()

pruebas()

File: repos_estudiantes_ALVAREZ_tercera_entrega/CHINO_BLANCO_NOEMI/Generacidad/Ejercicio3/Pila.java
package Generacidad.Ejercicio3;


import java.util.ArrayList;

public class Pila<T> {
    private ArrayList<T> elementos;

    public Pila() {
        this.elementos = new ArrayList<>();
    }

    // a) 
    public void apilar(T elemento) {
        elementos.add(elemento);
    }

    // b) 
    public T desapilar() {
        if (elementos.isEmpty()) {
            return null; 
        }
        return elementos.remove(elementos.size() - 1);
    }

    // d) 
    public void mostrar() {
        System.out.println("Contenido de la pila:");
        for (int i = elementos.size() - 1; i >= 0; i--) {
            System.out.println(elementos.get(i));
        }
    }

   
    public boolean estaVacia() {
        return elementos.isEmpty();
    }

    public static void main(String[] args) {
        System.out.println("Pila de enteros");
        Pila<Integer> pilaEnteros = new Pila<>();
        pilaEnteros.apilar(10);
        pilaEnteros.apilar(20);
        pilaEnteros.apilar(30);
        pilaEnteros.mostrar();
        System.out.println("Desapilar: " + pilaEnteros.desapilar());
        pilaEnteros.mostrar();

        System.out.println("\nPila de cadenas");
        Pila<String> pilaStrings = new Pila<>();
        pilaStrings.apilar("noemi chino blanco");
        pilaStrings.apilar("no");
        pilaStrings.mostrar();
        System.out.println("Desapilar: " + pilaStrings.desapilar());
        pilaStrings.mostrar();
    }
}

